### PR TITLE
Fix publish hostname in china and gov clouds

### PR DESCRIPTION
**Pull request by maddieford**
## Description
Publish hostname is failing in china and gov clouds because the domain names in these clouds are different than Public. 
The test was checking for the public cloud domain name (internal.cloudapp.net), but the domain names in China and Gov are the following:

- internal.chinacloudapp.cn
- internal.usgovcloudapp.net

Updated the regex to support any of these domain names

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests_e2e/tests/publish_hostname/publish_hostname.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def add_vm_password(self):
 
     def check_and_install_dns_tools(self):
         lookup_cmd = "dig -x {0}".format(self._private_ip)
-        dns_regex = r"[\S\s]*;; ANSWER SECTION:\s.*PTR\s*(?P<hostname>.*).internal.cloudapp.net.[\S\s]*"
+        dns_regex = r"[\S\s]*;; ANSWER SECTION:\s.*PTR\s*(?P<hostname>.*).internal.*[\S\s]*"
```
Comment (narrieta) [inline tests_e2e/tests/publish_hostname/publish_hostname.py:58]: should we use "internal\.(cloudapp\.net|chinacloudapp\.cn|usgovcloudapp.net)" instead?

File: tests_e2e/tests/publish_hostname/publish_hostname.py [hunk 2]
```diff
@@ -55,7 +55,7 @@ def add_vm_password(self):
 
     def check_and_install_dns_tools(self):
         lookup_cmd = "dig -x {0}".format(self._private_ip)
-        dns_regex = r"[\S\s]*;; ANSWER SECTION:\s.*PTR\s*(?P<hostname>.*).internal.cloudapp.net.[\S\s]*"
+        dns_regex = r"[\S\s]*;; ANSWER SECTION:\s.*PTR\s*(?P<hostname>.*).internal.(cloudapp.net|chinacloudapp.cn|usgovcloudapp.net).*[\S\s]*"
```
Comment (narrieta) [inline tests_e2e/tests/publish_hostname/publish_hostname.py:58]: minor: you should escape the dot operator: "\.internal\."

Comment (narrieta) [reply tests_e2e/tests/publish_hostname/publish_hostname.py:58]: seems like the backslash was not visible in my previous comment (need to be escaped in the comment :))

Comment (maddieford) [reply tests_e2e/tests/publish_hostname/publish_hostname.py]: Updated :)

**Review by narrieta — APPROVED** [review]

**Review by nagworld9 — APPROVED** [review]

**Pull request merged**